rnd/autogpt_server/autogpt_server/util/settings.py

```diff
@@ -74,7 +73,13 @@ def settings_customise_sources(
         dotenv_settings: PydanticBaseSettingsSource,
         file_secret_settings: PydanticBaseSettingsSource,
     ) -> Tuple[PydanticBaseSettingsSource, ...]:
-        return (JsonConfigSettingsSource(settings_cls),)
+        return (
+            env_settings,
+            file_secret_settings,
+            dotenv_settings,
+            init_settings,
+            JsonConfigSettingsSource(settings_cls),
```

Comment:
[The order of the returned callables decides the priority of inputs; first item is the highest priority.](https://docs.pydantic.dev/latest/concepts/pydantic_settings/#changing-priority)

So I'd suggest swapping `init_settings` with `JsonConfigSettingsSource(settings_cls)` so the json file can override default settings.